### PR TITLE
Using PInvoke for Parquet cpp library

### PR DESCRIPTION
**Pull request by moria97**
Current c++/cli implementation is limited to run in windows platform only. Change to PInvoke implementation for cross platform compatibility.
Changes include:
- Use vcpkg to manage arrow dependencies
- Build cross platform parquet native lib using CMake.
- Update CI pipelines for building native nugets.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs [hunk 1]
```diff
@@ -366,8 +366,14 @@ private async Task RenewJobLockLeaseAsync(CancellationToken cancellationToken =
...
             {
-                _logger.LogError("Failed to renew job lease.");
-                throw;
+                if (_renewLockTimer.Enabled)
+                {
+                    _logger.LogError("Failed to renew job lease.");
+                    throw;
+                }
+
+                _logger.LogInformation("Refresh timer already stopped.");
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs:375]: When will this case happen? In future distributed system?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaManager.cs [hunk 2]
```diff
@@ -83,9 +90,74 @@ public FhirParquetSchemaNode GetSchema(string schemaType)
...
+        {
+            return _resourceSchemaNodesMap.ToDictionary(pair => pair.Key, pair => Newtonsoft.Json.JsonConvert.SerializeObject(pair.Value));
+        }
+
         public Dictionary<string, FhirParquetSchemaNode> GetAllSchemas()
         {
             return new Dictionary<string, FhirParquetSchemaNode>(_resourceSchemaNodesMap);
         }
+
+        private Dictionary<string, FhirParquetSchemaNode> LoadSchemas(string schemaDirectoryPath)
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaManager.cs:103]: Currently the local default schemas have become embedded resource?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json [hunk 3]
```diff
@@ -16,6 +16,9 @@
     "requiredTypes": "",
     "typeFilters": ""
   },
+  "schema": {
+    "schemaCollectionDirectory": "./schemas"
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json:20]: Maybe we don't need schema directory path

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs [hunk 4]
```diff
@@ -123,7 +123,8 @@ public async Task GivenAValidInputData_WhenProcessWithCustomizedSchema_CorrectRe
         public async Task GivenAValidMultipleLargeInputData_WhenProcess_CorrectResultShouldBeReturned()
         {
             var largePatientSingleSet = TestUtils.LoadNdjsonData(Path.Combine(TestUtils.TestDataFolder, "Large_Patient.ndjson"));
-            var largeTestData = Enumerable.Repeat(largePatientSingleSet, 100).SelectMany(x => x);
+            // Maximum row number for arrow cpp parser is 100000, see https://github.com/apache/arrow/blob/42a9b32141c3c5a7178bef6644872d14f3051ce6/cpp/src/arrow/json/parser.h#L51
+            var largeTestData = Enumerable.Repeat(largePatientSingleSet, 10).SelectMany(x => x);
```
Comment (QuanWanxx) [inline FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs:127]: Why the parquet conversion was successfully before? Is this a linux limitation?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Microsoft.Health.Fhir.Synapse.FunctionApp.csproj [hunk 5]
```diff
@@ -12,6 +12,7 @@
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Timer" Version="4.0.1" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.0.3" OutputItemType="Analyzer" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.1.0" />
+    <PackageReference Include="Microsoft.Health.Fhir.Synapse.Parquet" Version="1.0.0.41" />
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Microsoft.Health.Fhir.Synapse.FunctionApp.csproj:15]: Nit: Maybe we don't need to include the Parquet lib here.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir.Synapse.Tool.csproj [hunk 6]
```diff
@@ -11,6 +11,7 @@
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Health.Fhir.Synapse.Parquet" Version="1.0.0.41" />
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir.Synapse.Tool.csproj:14]: Nit: Maybe we don't need to include the Parquet lib here.?

File: FhirToDataLake/azurepipelines/pack-nugets.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    mkdir -p src/Microsoft.Health.Fhir.Synapse.Parquet/native/windows-x64
+    cp -r $(Pipeline.Workspace)/native-linux-x64/** src/Microsoft.Health.Fhir.Synapse.Parquet/native/linux-x64
+    cp -r $(Pipeline.Workspace)/native-windows-x64/** src/Microsoft.Health.Fhir.Synapse.Parquet/native/windows-x64
+    dotnet pack src/Microsoft.Health.Fhir.Synapse.Parquet/Microsoft.Health.Fhir.Synapse.Parquet.csproj -p:NuspecFile=BuildParquet.nuspec -p:NuspecProperties="version=$(version)" -c Release -o '$(Build.ArtifactStagingDirectory)/nupkgs'
+  workingDirectory: FhirToDataLake
+  displayName: Nuget pack
+
+- task: PublishBuildArtifacts@1
+  inputs:
+    pathToPublish: '$(Build.ArtifactStagingDirectory)/nupkgs'
```
Comment (QuanWanxx) [inline FhirToDataLake/azurepipelines/pack-nugets.yml:21]: Just a discussion, maybe we can move Parquet project in more separated location? Now the other codes refer Parquet Nuget package, feel like it should be placed in separated location, and be managed alone.

File: FhirToDataLake/Microsoft.Health.Fhir.Synapse.sln [hunk 8]
```diff
@@ -34,14 +30,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Fhir.Synap
...
-EndProject
-Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Microsoft.Health.Fhir.Synapse.Parquet.Native.UnitTests", "test\Microsoft.Health.Fhir.Synapse.Parquet.Native.UnitTests\Microsoft.Health.Fhir.Synapse.Parquet.Native.UnitTests.vcxproj", "{9AA63275-45BE-4019-B5F1-E371A985D045}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Fhir.Synapse.Core.UnitTests", "test\Microsoft.Health.Fhir.Synapse.Core.UnitTests\Microsoft.Health.Fhir.Synapse.Core.UnitTests.csproj", "{8805CF56-8937-4FF9-8CF0-3450258C9DE6}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Fhir.Synapse.E2ETests", "test\Microsoft.Health.Fhir.Synapse.E2ETests\Microsoft.Health.Fhir.Synapse.E2ETests.csproj", "{01C19DA0-63E6-43E2-B421-12CC8BD7EB94}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Health.Fhir.Synapse.Parquet.UnitTests", "test\Microsoft.Health.Fhir.Synapse.Parquet.UnitTests\Microsoft.Health.Fhir.Synapse.Parquet.UnitTests.csproj", "{35755AAC-C867-499E-8F49-A141ADDD54FA}"
```
Comment (hongyh13) [inline FhirToDataLake/Microsoft.Health.Fhir.Synapse.sln:37]: Is it properly to add `Parquet.unitTests` here?

File: azure-pipelines.yml [hunk 9]
```diff
@@ -37,7 +39,7 @@ stages:
   jobs:
```
Comment (hongyh13) [inline azure-pipelines.yml:39]: What's the consideration that `FhirToDataLake` is built on ubuntu, while it is released on windows?

File: azure-pipelines.yml [hunk 10]
```diff
@@ -37,7 +39,7 @@ stages:
...
   - job: SynapseE2ETest
     steps:
-    - template: FhirToDataLake/synapse-e2e-test.yml
+    - template: FhirToDataLake/azurepipelines/synapse-e2e-test.yml
 
   - job: PublishFhirToCdm
```
Comment (hongyh13) [inline azure-pipelines.yml:44]: just curious: why `PublishFhirToCdm` dependsOn `SynapseE2ETest`?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Parquet/Microsoft.Health.Fhir.Synapse.Parquet.csproj [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Parquet/Microsoft.Health.Fhir.Synapse.Parquet.csproj:1]: A new folder for `parquet` project?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs [hunk 12]
```diff
@@ -17,7 +17,8 @@
 using Microsoft.Health.Fhir.Synapse.Common.Models.Data;
 using Microsoft.Health.Fhir.Synapse.Core.DataProcessor.DataConverter;
 using Microsoft.Health.Fhir.Synapse.Core.Exceptions;
-using Microsoft.Health.Fhir.Synapse.Parquet.CLR;
+using Microsoft.Health.Fhir.Synapse.Parquet;
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:20]: Is neccessary to locate `Parquet` under the namespace `Synapse`?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs [hunk 13]
```diff
@@ -48,18 +49,11 @@ public sealed class ParquetDataProcessor : IColumnDataProcessor
...
-            catch (Exception ex)
-            {
-                _logger.LogError("Create ParquetConverterWrapper failed.");
-                throw new ParquetDataProcessorException("Create ParquetConverterWrapper failed. " + ex.Message, ex);
-            }
-
-            _logger.LogInformation($"ParquetDataProcessor initialized successfully with ArrowConfiguration: {JsonConvert.SerializeObject(arrowConfiguration.Value)}.");
+            var schemaSet = fhirSchemaManager.GetAllSchemaContent();
+            _parquetConverter.InitializeSchemaSet(schemaSet);
+            _logger.LogInformation($"ParquetDataProcessor initialized successfully with {schemaSet.Count()} parquet schemas.");
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:56]: Is it possible that parquet converter throws an exception while initializing schema set?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs [hunk 14]
```diff
@@ -366,8 +366,14 @@ private async Task RenewJobLockLeaseAsync(CancellationToken cancellationToken =
             }
             catch
             {
-                _logger.LogError("Failed to renew job lease.");
-                throw;
+                if (_renewLockTimer.Enabled)
+                {
+                    _logger.LogError("Failed to renew job lease.");
+                    throw;
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs:372]: who will handle the exception? what's the expected behavior for it?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs [hunk 15]
```diff
@@ -366,8 +366,14 @@ private async Task RenewJobLockLeaseAsync(CancellationToken cancellationToken =
...
+                }
+
+                _logger.LogInformation("Refresh timer already stopped.");
+                return;
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs:376]: how the caller handle this case that renew locker fails while no exception is threw?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json [hunk 16]
```diff
@@ -16,6 +16,9 @@
     "requiredTypes": "",
     "typeFilters": ""
   },
+  "schema": {
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json:19]: duplicate key `schema` here

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs [hunk 17]
```diff
@@ -123,6 +123,7 @@ public async Task GivenAValidInputData_WhenProcessWithCustomizedSchema_CorrectRe
         public async Task GivenAValidMultipleLargeInputData_WhenProcess_CorrectResultShouldBeReturned()
         {
             var largePatientSingleSet = TestUtils.LoadNdjsonData(Path.Combine(TestUtils.TestDataFolder, "Large_Patient.ndjson"));
+            // Maximum row number for arrow cpp parser is 100000, see https://github.com/apache/arrow/blob/42a9b32141c3c5a7178bef6644872d14f3051ce6/cpp/src/arrow/json/parser.h#L51
```
Comment (hongyh13) [inline FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs:126]: add a blank line here

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaManager.cs (hunk 2 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaManager.cs]: Thanks. Removed

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs (hunk 4 above)
Comment (moria97) [reply FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs]: Because the block size used in earlier tests are default value, which is small and should not reach the limit

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir.Synapse.Tool.csproj (hunk 6 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/Microsoft.Health.Fhir.Synapse.Tool.csproj]: We need the lib here to make `copy native libs` happen during nuget restore phase

File: FhirToDataLake/azurepipelines/pack-nugets.yml (hunk 7 above)
Comment (moria97) [reply FhirToDataLake/azurepipelines/pack-nugets.yml:21]: Good advice, removed all native lib related code to a seperate folder

File: FhirToDataLake/Microsoft.Health.Fhir.Synapse.sln (hunk 8 above)
Comment (moria97) [reply FhirToDataLake/Microsoft.Health.Fhir.Synapse.sln]: Removed. Thanks

File: azure-pipelines.yml (hunk 9 above)
Comment (moria97) [reply azure-pipelines.yml:39]: Actually this PR mainly covers the build pipeline. Changes need be made in future to fix the release pipeline. Both pipelines should run in Linux platform.

File: azure-pipelines.yml (hunk 10 above)
Comment (moria97) [reply azure-pipelines.yml:44]: just want to make sure the E2E tests passed before releasing~

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Parquet/Microsoft.Health.Fhir.Synapse.Parquet.csproj (hunk 11 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Parquet/Microsoft.Health.Fhir.Synapse.Parquet.csproj]: Sure. Created new folder for parquet projects

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs (hunk 12 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:20]: Fixed. Thanks

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs (hunk 13 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:55]: Yes. It will throw when initializing with given schema failed

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs (hunk 15 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs:376]: This is some corner cases when the timer stops but the renew request was not cancelled

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json (hunk 16 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Tool/appsettings.json]: removed. thanks

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs (hunk 17 above)
Comment (moria97) [reply FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/DataProcessor/ParquetDataProcessorTests.cs]: Added. Thanks

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Microsoft.Health.Fhir.Synapse.FunctionApp.csproj (hunk 5 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.FunctionApp/Microsoft.Health.Fhir.Synapse.FunctionApp.csproj]: We need the lib here to make copy native libs happen during nuget restore phase

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs [hunk 18]
```diff
@@ -17,7 +17,8 @@
 using Microsoft.Health.Fhir.Synapse.Common.Models.Data;
 using Microsoft.Health.Fhir.Synapse.Core.DataProcessor.DataConverter;
 using Microsoft.Health.Fhir.Synapse.Core.Exceptions;
-using Microsoft.Health.Fhir.Synapse.Parquet.CLR;
+using Microsoft.Health.Fhir.Synapse.Parquet;
+
```
[the remark this answers is not in the document]
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs]: Removed. Thanks

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs [hunk 19]
```diff
@@ -48,18 +49,11 @@ public sealed class ParquetDataProcessor : IColumnDataProcessor
...
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError("Create ParquetConverterWrapper failed.");
-                throw new ParquetDataProcessorException("Create ParquetConverterWrapper failed. " + ex.Message, ex);
-            }
-
-            _logger.LogInformation($"ParquetDataProcessor initialized successfully with ArrowConfiguration: {JsonConvert.SerializeObject(arrowConfiguration.Value)}.");
+            var schemaSet = fhirSchemaManager.GetAllSchemaContent();
+            _parquetConverter.InitializeSchemaSet(schemaSet);
```
[the remark this answers is not in the document]
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:54]: Good idea. Fixed

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs (hunk 1 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs:375]: Corner cases found in E2E tests

File: .gitmodules [hunk 20]
```diff
@@ -0,0 +1,3 @@
+[submodule "FhirToDataLake/native/parquet/cpp/vcpkg"]
+	path = FhirToDataLake/native/parquet/cpp/vcpkg
```
Comment (ACMoretxj) [inline .gitmodules:2]: Do we need a `branch` option here to track the dedicated contents? Or it is meant to always track the latest `HEAD`?

File: FhirToDataLake/azurepipelines/ci.yml [hunk 21]
```diff
@@ -0,0 +1,106 @@
...
+  displayName: 'NuGet Authenticate'
+
+- task: NuGetToolInstaller@1
+  displayName: 'Use NuGet 5.11'
+  inputs:
+    versionSpec: 5.11
+
+# Install for Azure Function dependencies
+- task: UseDotNet@2
+  displayName: 'Install .NET Core sdk 3.1'
```
Comment (ACMoretxj) [inline FhirToDataLake/azurepipelines/ci.yml:12]: Also needs to upgrade to `.NET 6.0`?

File: FhirToDataLake/azurepipelines/ci.yml [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+- task: DotNetCoreCLI@2
+  displayName: 'dotnet build solution'
+  inputs:
+    command: build
+    projects: FhirToDataLake/*.sln
+    arguments: '--configuration $(buildConfiguration) -p:Version=$(version)'
+
+- script: |
+    docker pull mcr.microsoft.com/azure-storage/azurite
+    docker run -d --name azurite-emulator-analytics-ci -p 10000:10000 -p 10001:10001 -p 10002:10002 \
```
Comment (ACMoretxj) [inline FhirToDataLake/azurepipelines/ci.yml:43]: Maybe using `-p 10000-10002:10000:10002` for simplification.

File: .gitmodules (hunk 20 above)
Comment (moria97) [reply .gitmodules:2]: It will use the [commit](https://github.com/microsoft/FHIR-Analytics-Pipelines/pull/134/files#diff-f7552bb6842d6fb24280f9b371728329052fba18841231b57a23b7b312b36701) so the dependencies versions can be fixed.

File: FhirToDataLake/azurepipelines/nativeci.yml [hunk 23]
```diff
@@ -0,0 +1,34 @@
+name: $(Date:yyyyMMdd)$(Rev:.r)
+
+variables:
```
Comment (hongyh13) [inline FhirToDataLake/azurepipelines/nativeci.yml:3]: how about creating a sub folder for native yaml files?

File: FhirToDataLake/azurepipelines/ci.yml (hunk 22 above)
Comment (moria97) [reply FhirToDataLake/azurepipelines/ci.yml]: Updated. Thanks

File: FhirToDataLake/azurepipelines/ci.yml (hunk 21 above)
Comment (moria97) [reply FhirToDataLake/azurepipelines/ci.yml:12]: This is just for building function app dependencies in build agent. Will remove when we deprecate the functionapp project

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs (hunk 13 above)
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:54]: so, will the exception be thrown by native code, and be caught by .net code? where the exception will be caught? need we add unit tests to make sure it？

File: azure-pipelines.yml [hunk 24]
```diff
@@ -37,7 +39,7 @@ stages:
   jobs:
   - job: SynapseE2ETest
```
[the remark this answers is not in the document]
Comment (moria97) [reply azure-pipelines.yml:40]: Good idea. Added this suggestion in the [ci update pr](https://github.com/microsoft/FHIR-Analytics-Pipelines/pull/142)

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs (hunk 13 above)
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs:54]: The native code will not throw exception. It will return status code. Exceptions will be thrown from managed code. We have tests for [this](https://github.com/microsoft/FHIR-Analytics-Pipelines/blob/2e1c044addf629133b890c3cac6c92b77110895a/FhirToDataLake/native/parquet/csharp/test/ParquetConverterTests.cs#L26)

**Review by hongyh13 — APPROVED** [review]

File: FhirToDataLake/native/parquet/csharp/src/BuildParquet.nuspec [hunk 25]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <metadata>
+        <id>Microsoft.Health.Parquet</id>
+        <version>1.0.0.0</version>
```
[the remark this answers is not in the document]
Comment (moria97) [reply FhirToDataLake/native/parquet/csharp/src/BuildParquet.nuspec:5]: This is just a placeholder, the version will be replaced by build variables in ci pipeline

File: FhirToDataLake/native/parquet/csharp/src/Microsoft.Health.Parquet.csproj [hunk 26]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (moria97) [reply FhirToDataLake/native/parquet/csharp/src/Microsoft.Health.Parquet.csproj]: Sure. Updated.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/IFhirSchemaManager.cs [hunk 27]
```diff
@@ -24,6 +24,8 @@ public interface IFhirSchemaManager<T>
         /// <returns>Instance of FhirSchemaNode, represents the schema for given resource type.</returns>
         public T GetSchema(string schemaType);
 
+        public Dictionary<string, string> GetAllSchemaContent();
```
[the remark this answers is not in the document]
Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/IFhirSchemaManager.cs:32]: Thanks. Updated

File: FhirToDataLake/azurepipelines/nativeci.yml (hunk 23 above)
Comment (moria97) [reply FhirToDataLake/azurepipelines/nativeci.yml]: Thanks. updated
